File: leetcode/easy/349_两个数组的交集/349_两个数组的交集.py
# 哈希集合法
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        result_set = set()
        nums_set = set(nums1)
        for num in nums2:
            if num in nums_set:
                result_set.add(num)
        return list(result_set)

File: leetcode/easy/349_两个数组的交集/349_两个数组的交集.go
//哈希集合法
func intersection(nums1 []int, nums2 []int) []int {
    resultSet := make(map[int]bool)
    numsSet := make(map[int]bool)
    
    // 将nums1放入哈希集合
    for _, num := range nums1 {
        numsSet[num] = true
    }
    
    // 遍历nums2，查找交集
    for _, num := range nums2 {
        if numsSet[num] {
            resultSet[num] = true
        }
    }
    
    // 转换为切片返回
    result := make([]int, 0, len(resultSet))
    for num := range resultSet {
        result = append(result, num)
    }
    return result
}